src/golang/lib/workflow/orchestrator/orchestrator.go

```diff
@@ -144,14 +155,15 @@ func opFailureError(failureType shared.FailureType, op operator.Operator) error
 
 func (orch *aqOrchestrator) execute(
 	ctx context.Context,
-	dag dag.WorkflowDag,
 	timeConfig *AqueductTimeConfig,
 	jobManager job.JobManager,
 	shouldPersistResults bool,
 ) error {
 	// These are the operators of immediate interest. They either need to be scheduled or polled on.
 	inProgressOps := orch.inProgressOps
 	completedOps := orch.completedOps
+	dag := orch.dag
+	opToDependencyCount := orch.opToDependencyCount
 
 	// Kick off execution by starting all operators that don't have any inputs.
 	for _, op := range dag.Operators() {
```

Comment:
I think we could also use `opToDependencyCount` here to avoid sending request for every `artf->op` edge